cpp/include/cudf/io/orc.hpp

```diff
@@ -0,0 +1,832 @@
+
+  // Whether to convert decimals to float64
+  bool _decimals_as_float64 = true;
+  // For decimals as int, optional forced decimal scale;
+  // -1 is auto (column scale), >=0: number of fractional digits
+  size_type _forced_decimals_scale = -1;
+
+  friend orc_reader_options_builder;
+
+  /**
+   * @brief Constructor from source info.
+   *
+   * @param src source information used to read orc file.
+   */
+  explicit orc_reader_options(source_info const& src) : _source(src) {}
+
+ public:
+  /**
+   * @brief Default constructor.
+   *
+   * This has been added since Cython requires a default constructor to create objects on stack.
+   */
+  orc_reader_options() = default;
+
+  /**
+   * @brief Creates `orc_reader_options_builder` which will build `orc_reader_options`.
+   *
+   * @param src Source information to read orc file.
+   * @return Builder to build reader options.
+   */
+  static orc_reader_options_builder builder(source_info const& src);
+
+  /**
+   * @brief Returns source info.
+   */
+  source_info const& get_source() const { return _source; }
+
+  /**
+   * @brief Returns names of the columns to read.
+   */
+  std::vector<std::string> const& get_columns() const { return _columns; }
+
+  /**
+   * @brief Returns list of individual stripes to read.
+   */
+  std::vector<size_type> const& get_stripes() const { return _stripes; }
+
+  /**
+   * @brief Returns number of rows to skip from the start.
+   */
+  size_type get_skip_rows() const { return _skip_rows; }
+
+  /**
+   * @brief Returns number of row to read.
+   */
+  size_type get_num_rows() const { return _num_rows; }
+
+  /**
+   * @brief Whether to use row index to speed-up reading.
+   */
+  bool is_enabled_use_index() const { return _use_index; }
+
+  /**
+   * @brief Whether to use numpy-compatible dtypes.
+   */
+  bool is_enabled_use_np_dtypes() const { return _use_np_dtypes; }
+
+  /**
+   * @brief Returns timestamp type to which timestamp column will be cast.
+   */
+  data_type get_timestamp_type() const { return _timestamp_type; }
+
+  /**
+   * @brief Whether to convert decimals to float64.
+   */
+  bool is_enabled_decimals_as_float64() const { return _decimals_as_float64; }
+
+  /**
+   * @brief Returns whether decimal scale is inferred or forced to have limited fractional digits.
+   */
+  size_type get_forced_decimals_scale() const { return _forced_decimals_scale; }
+
+  // Setters
+
+  /**
+   * @brief Sets names of the column to read.
+   *
+   * @param col_names Vector of column names.
+   */
+  void set_columns(std::vector<std::string> col_names) { _columns = std::move(col_names); }
+
+  /**
+   * @brief Sets list of individual stripes to read.
+   *
+   * @param stripes Vector of individual stripes.
+   */
+  void set_stripes(std::vector<size_type> stripes)
+  {
+    CUDF_EXPECTS(stripes.empty() or (_skip_rows == 0 and _num_rows == -1),
+                 "Can't set both stripes along with skip_rows/num_rows");
```

Comment:
This would improve the error message
```suggestion
    CUDF_EXPECTS(stripes.empty() or (_skip_rows == 0), "Can't set stripes along with skip_rows");
    CUDF_EXPECTS(stripes.empty() or (_num_rows == -1),  "Can't set stripes along with num_rows");
```